src/dmn/trn_agent_xdp_usr.c

```diff
@@ -251,12 +251,51 @@ int trn_agent_delete_endpoint(struct agent_user_metadata_t *umd,
 	return 0;
 }
 
+int trn_agent_update_packet_metadata(struct agent_user_metadata_t *umd,
+			      struct packet_metadata_key_t *key,
+			      struct packet_metadata_t *packet_metadata)
+{
+	int err = bpf_map_update_elem(umd->packet_metadata_map_fd, key, packet_metadata, 0);
+	if (err) {
+		TRN_LOG_ERROR("Store packet metadata mapping failed (err:%d).", err);
+		return 1;
+	}
+	return 0;
+}
+
+int trn_agent_get_packet_metadata(struct agent_user_metadata_t *umd,
+			   struct packet_metadata_key_t *key, struct packet_metadata_t *packet_metadata)
+{
+	int err = bpf_map_lookup_elem(umd->packet_metadata_map_fd, key, packet_metadata);
+	if (err) {
+		TRN_LOG_ERROR(
+			"Querying agent packet metadata mapping failed (err:%d).",
+			err);
+		return 1;
+	}
+	return 0;
+}
+
+int trn_agent_delete_packet_metadata(struct agent_user_metadata_t *umd,
+			      struct packet_metadata_key_t *key)
+{
+	int err = bpf_map_delete_elem(umd->packet_metadata_map_fd, key);
+	if (err) {
```

Comment:
If error is "not found", do we still want to log error message?

Reply:
Yes. When there is request to delete the entry, it assume the entry is there.